src/main/java/net/tlabs/tablesaw/parquet/TablesawParquetReader.java

```diff
@@ -33,32 +33,46 @@
 
 public class TablesawParquetReader implements DataReader<TablesawParquetReadOptions> {
 
-  private static final Logger LOG = LoggerFactory.getLogger(TablesawParquetReader.class);
+    private static final Logger LOG = LoggerFactory.getLogger(TablesawParquetReader.class);
 
-  @Override
-  public Table read(final Source source) throws IOException {
-    final File file = source.file();
-    if (file != null) {
-      return read(TablesawParquetReadOptions.builder(file).build());
+    /**
+     * Registers the TablesawParquetReader in the default tablesaw registry.
+     * Reader is associated with its options and the '.parquet' extension.
+     */
+    public static void register() {
+        final TablesawParquetReader registeredInstance = new TablesawParquetReader();
+        Table.defaultReaderRegistry.registerOptions(TablesawParquetReadOptions.class, registeredInstance);
+        Table.defaultReaderRegistry.registerExtension("parquet", registeredInstance);
     }
-    throw new UnsupportedOperationException(
-        "Can only work with file based source, please use the read(TablesawParquetReadOptions) method for additional possibilities");
-  }
 
-  @Override
-  public Table read(final TablesawParquetReadOptions options) throws IOException {
-    final long start = System.currentTimeMillis();
-    final String inputPath = options.getInputPath();
-    final Path path = new Path(inputPath);
-    final TablesawReadSupport readSupport = new TablesawReadSupport(options);
-    try (final ParquetReader<Row> reader = ParquetReader.<Row>builder(readSupport, path).build()) {
-      int i = 0;
-      while (reader.read() != null) {
-        i++;
-      }
-      final long end = System.currentTimeMillis();
-      LOG.debug("Finished reading {} rows from {} in {} ms", i, inputPath, (end - start));
+    public TablesawParquetReader() {
+        super();
+    }
+
+    @Override
+    public Table read(final Source source) throws IOException {
+        final File file = source.file();
+        if (file != null) {
```

Comment:
what happens if file.isDirectory() ?

Reply:
Good question. The underlying parquet reader from parquet-mr supports partitioned datasets, i.e. the table is partitioned by a key in several individual parquet files. If the file is a directory it will try to assemble the different parquet files in a single dataset (and throw some IOException if it cannot).

I did not test this (need to generate test files somehow), but it is entirely manager by parquet-mr so I am pretty confident it should work out of the box.